hibernate-core/src/main/java/org/hibernate/stat/internal/StatisticsImpl.java

```diff
@@ -540,18 +551,29 @@ public String[] getSecondLevelCacheRegionNames() {
 
 	@Override
 	public CacheRegionStatisticsImpl getDomainDataRegionStatistics(String regionName) {
-		return l2CacheStatsMap.getOrCompute(
-				regionName,
-				this::instantiateCacheRegionStatistics
+		return NullnessUtil.castNonNull(
+					l2CacheStatsMap.getOrCompute(
+						regionName,
+						this::instantiateCacheRegionStatistics
+					)
 		);
 	}
 
 	@Override
-	public CacheRegionStatisticsImpl getQueryRegionStatistics(final String regionName) {
-		return l2CacheStatsMap.getOrCompute( regionName, this::computeQueryRegionStatistics );
+	public @Nullable CacheRegionStatisticsImpl getQueryRegionStatistics(final String regionName) {
+		return l2CacheStatsMap.getOrCompute(
+				regionName,
+
+				new Function<>() {
```

Comment:
Is this really necessary? Other method references seem to work fine as well.

Reply:
Without it, checkerframework complains, it needs the return type of those particular functions to be @Nullable, and that's the only way I could achieve that, which doesn't mean there aren't better ways, I just don't know them ;)

Reply:
Alright, if that is what it takes, fine then.